src/core/project.js

```diff
@@ -24,16 +24,35 @@ export type ProjectId = string;
  * the future (e.g. showing the last update time for each of the project's data
  * dependencies).
  */
-export type Project = {|
+export type Project = Project_v040;
+export type SupportedProject = Project_v030 | Project_v031 | Project_v040;
+
+type Project_v040 = {|
   +id: ProjectId,
   +repoIds: $ReadOnlyArray<RepoId>,
   +discourseServer: DiscourseServer | null,
   +identities: $ReadOnlyArray<Identity>,
 |};
 
+type Project_v031 = {|
+  ...Project_v040,
+  +discourseServer: {|
+    +serverUrl: string,
+    +apiUsername?: string,
+  |} | null,
+|};
+
+type Project_v030 = {|
```

Comment:
Looks like Project_v030 is the same as Project_v031? Any reason we don't just say `type Project_v030 = Project_v031`?

Reply:
Difference is `discourseServer.apiUsername` became optional.